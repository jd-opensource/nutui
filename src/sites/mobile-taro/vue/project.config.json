{
  "miniprogramRoot": "./dist",
  "projectname": "@nutui/nutui-taro-mobile",
  "description": "nutui-taro-vue",
  "appid": "touristappid",
  "setting": {
    "urlCheck": true,
    "es6": false,
    "postcss": false,
    "minified": false
  },
  "compileType": "miniprogram"
}
