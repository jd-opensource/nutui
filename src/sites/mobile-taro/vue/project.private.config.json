{
  "setting": {},
  "condition": {
    "plugin": {
      "list": []
    },
    "game": {
      "list": []
    },
    "gamePlugin": {
      "list": []
    },
    "miniprogram": {
      "list": [
        {
          "name": "business/pages/address/index",
          "pathName": "business/pages/address/index",
          "query": "",
          "scene": null
        },
        {
          "name": "pages/index/index",
          "pathName": "pages/index/index",
          "query": "",
          "scene": null
        },
        {
          "name": "feedback/pages/noticebar/index",
          "pathName": "feedback/pages/noticebar/index",
          "query": "",
          "scene": null
        },
        {
          "name": "feedback/pages/collapse/index",
          "pathName": "feedback/pages/collapse/index",
          "query": "",
          "scene": null
        },
        {
          "name": "feedback/pages/progress/index",
          "pathName": "feedback/pages/progress/index",
          "query": "",
          "scene": null
        },
        {
          "name": "feedback/pages/swiper/index",
          "pathName": "feedback/pages/swiper/index",
          "query": "",
          "scene": null
        }
      ]
    }
  }
}