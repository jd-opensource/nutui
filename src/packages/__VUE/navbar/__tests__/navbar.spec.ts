import { mount } from '@vue/test-utils';
import { nextTick } from 'vue';
import NavBar from '../index.vue';

test('Navbar: slot content & left & right', () => {
  const wrapper = mount(NavBar, {
    slots: {
      content: () => 'Title',
      left: () => 'Left',
      right: () => 'Right'
    }
  });
  expect(wrapper.find('.nut-navbar__title').text()).toBe('Title');
  expect(wrapper.find('.nut-navbar__left').text()).toBe('Left');
  expect(wrapper.find('.nut-navbar__right').text()).toBe('Right');
});

test('Navbar: prop left-text & desc', () => {
  const wrapper = mount(NavBar, {
    props: {
      leftText: 'left',
      desc: 'desc'
    }
  });

  const label1 = wrapper.find('.nut-navbar__left .nut-navbar__text');
  expect(label1.text()).toBe('left');
  const label2 = wrapper.find('.nut-navbar__right .nut-navbar__text');
  expect(label2.text()).toBe('desc');
});

test('Navbar: should render placeholder element when using placeholder prop', async () => {
  const wrapper = mount(NavBar, {
    props: {
      fixed: true,
      placeholder: true
    }
  });
  expect(wrapper.find('.nut-navbar--placeholder').html()).toMatchSnapshot();
});

test('Navbar: emit click-left & click-right', () => {
  const wrapper = mount(NavBar, {
    props: {
      titleIcon: true,
      title: '123'
    }
  });

  wrapper.find('.nut-navbar__left').trigger('click');
  expect(wrapper.emitted('clickBack')).toBeTruthy();
  wrapper.find('.nut-navbar__right').trigger('click');
  expect(wrapper.emitted('clickRight')).toBeTruthy();
  wrapper.find('.nut-navbar__title .title').trigger('click');
  expect(wrapper.emitted('clickTitle')).toBeTruthy();
  wrapper.find('.nut-navbar__title .icon').trigger('click');
  expect(wrapper.emitted('clickIcon')).toBeTruthy();
});

test('Navbar: should change z-index when using z-index prop', async () => {
  const wrapper = mount(NavBar, {
    props: {
      zIndex: 100
    }
  });
  await nextTick();
  const navbar = wrapper.find('.nut-navbar');
  expect(navbar.attributes('style')).includes('z-index: 100');
});
