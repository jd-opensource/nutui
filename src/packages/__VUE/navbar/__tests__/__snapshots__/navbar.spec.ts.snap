// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar: should render placeholder element when using placeholder prop 1`] = `
"<view class="nut-navbar--placeholder" style="height: auto;">
  <view class="nut-navbar nut-navbar--fixed" style="z-index: 10;">
    <view class="nut-navbar__left">
      <!--v-if-->
      <!--v-if-->
    </view>
    <view class="nut-navbar__title">
      <!--v-if-->
      <!--v-if-->
    </view>
    <view class="nut-navbar__right">
      <!--v-if-->
    </view>
  </view>
</view>"
`;
